composer/trainer/devices/device_gpu.py

```diff
@@ -33,14 +34,21 @@ def tensor_to_device(self, tensor: Tensor) -> Tensor:
     @contextmanager
     def precision_context(self, precision: Union[str, Precision]) -> Generator[None, None, None]:
         precision = Precision(precision)
+        enabled = False
         if precision == Precision.FP32:
             enabled = False
         elif precision == Precision.AMP:
             enabled = True
+        elif precision == Precision.BF16 and version.parse(torch.__version__) >= version.parse("1.10"):
+            enabled = True
         else:
             raise ValueError(f"Precision {precision} not supported for a GPU")
-        with torch.cuda.amp.autocast(enabled):  #type: ignore
-            yield
+        if precision == Precision.BF16:
+            with torch.cuda.amp.autocast(True, dtype=torch.bfloat16):
```

Comment:
maybe for reduced verbosity, we add a `dtype` variable defined in the above if/else statements, and just do:
```
with torch.cuda.amp.autocast(enabled, dtype=dtype)
    yield
```